Sources/SecureXPC/Client/XPCClient.swift

```diff
@@ -221,9 +222,24 @@ public class XPCClient {
         })
     }
 
-	// MARK: Abstract methods
+    private func getConnection() -> xpc_connection_t {
```

Comment:
If I'm understanding this correctly, once the connection gets invalidated (`XPC_ERROR_CONNECTION_INVALID`) the client becomes unusable. We shouldn't make users of this API have to manually handle this when we can automatically do so under the hood.

This is one of the major reasons the current design doesn't persist the connection (although as you note that of course that comes with a significant performance penalty) - by not doing so it always gracefully recovers from this situation. With the current design of this PR, I think it'd make sense to properly receive all of the XPC errors and then `nil` out the connection as needed.

Reply:
Ah yes, you're right. I need to pick off some parts of https://github.com/trilemma-dev/SecureXPC/pull/18/files to do this (introducing an error handler for the sake of the client's connection, not for the errors encounters on specific messages)